File: springboot/src/main/java/com/example/servidorcalendario/model/Evento.java
package com.example.servidorcalendario.model;

import jakarta.persistence.*;

@Entity
@Table(name = "eventos")
public class Evento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idEvento;

    private String titulo;
    private String descripcion;
    private String fechaInicio;
    private String fechaFin;
    private String ubicacion;
    private String estado; // PENDIENTE, CONFIRMADO
    private Integer idCalendario;

    public Evento() {}

    public Integer getIdEvento() { return idEvento; }
    public void setIdEvento(Integer idEvento) { this.idEvento = idEvento; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }

    public String getFechaInicio() { return fechaInicio; }
    public void setFechaInicio(String fechaInicio) { this.fechaInicio = fechaInicio; }

    public String getFechaFin() { return fechaFin; }
    public void setFechaFin(String fechaFin) { this.fechaFin = fechaFin; }

    public String getUbicacion() { return ubicacion; }
    public void setUbicacion(String ubicacion) { this.ubicacion = ubicacion; }

    public String getEstado() { return estado; }
    public void setEstado(String estado) { this.estado = estado; }

    public Integer getIdCalendario() { return idCalendario; }
    public void setIdCalendario(Integer idCalendario) { this.idCalendario = idCalendario; }
}

File: bocetoCalendario1_funcional/app/src/main/java/com/example/bocetocalendario1/datos/modelo/Evento.kt
package com.example.bocetocalendario1.datos.modelo

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "eventos",
    foreignKeys = [
        ForeignKey(
            entity = Calendario::class,
            parentColumns = ["id_calendario"],
            childColumns = ["id_calendario"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Evento(
    @PrimaryKey(autoGenerate = true) val id_evento: Int = 0,
    val titulo: String,
    val descripcion: String?,
    val fecha_inicio: String, // Formato "YYYY-MM-DD HH:MM:SS"
    val fecha_fin: String,
    val ubicacion: String?,
    val estado: String,     // 'PENDIENTE', 'CONFIRMADO'
    val id_calendario: Int
)